src/lib.rs

```diff
@@ -249,65 +275,68 @@ impl Dependencies {
         self.libs.iter().map(|(k, v)| (k.as_str(), v))
     }
 
-    fn aggregate_str<F: Fn(&Library) -> &Vec<String>>(
-        &self,
-        getter: F,
-    ) -> impl Iterator<Item = &str> {
-        self.libs
+    fn aggregate_str<F: Fn(&Library) -> &Vec<String>>(&self, getter: F) -> Vec<&str> {
+        let mut v = self
+            .libs
             .values()
             .map(|l| getter(l))
             .flatten()
             .map(|s| s.as_str())
-            .sorted()
-            .dedup()
+            .collect::<Vec<_>>();
+        v.sort();
+        v.dedup();
+        v
     }
 
-    fn aggregate_path_buf<F: Fn(&Library) -> &Vec<PathBuf>>(
-        &self,
-        getter: F,
-    ) -> impl Iterator<Item = &PathBuf> {
-        self.libs
+    fn aggregate_path_buf<F: Fn(&Library) -> &Vec<PathBuf>>(&self, getter: F) -> Vec<&PathBuf> {
+        let mut v = self
+            .libs
             .values()
             .map(|l| getter(l))
             .flatten()
-            .sorted()
-            .dedup()
+            .collect::<Vec<_>>();
+        v.sort();
+        v.dedup();
+        v
     }
 
     /// An iterator returning each [Library::libs] of each library, removing duplicates.
-    pub fn all_libs(&self) -> impl Iterator<Item = &str> {
+    pub fn all_libs(&self) -> Vec<&str> {
         self.aggregate_str(|l| &l.libs)
     }
 
     /// An iterator returning each [Library::link_paths] of each library, removing duplicates.
-    pub fn all_link_paths(&self) -> impl Iterator<Item = &PathBuf> {
+    pub fn all_link_paths(&self) -> Vec<&PathBuf> {
         self.aggregate_path_buf(|l| &l.link_paths)
     }
 
     /// An iterator returning each [Library::frameworks] of each library, removing duplicates.
-    pub fn all_frameworks(&self) -> impl Iterator<Item = &str> {
+    pub fn all_frameworks(&self) -> Vec<&str> {
         self.aggregate_str(|l| &l.frameworks)
     }
 
     /// An iterator returning each [Library::framework_paths] of each library, removing duplicates.
-    pub fn all_framework_paths(&self) -> impl Iterator<Item = &PathBuf> {
+    pub fn all_framework_paths(&self) -> Vec<&PathBuf> {
         self.aggregate_path_buf(|l| &l.framework_paths)
     }
 
     /// An iterator returning each [Library::include_paths] of each library, removing duplicates.
-    pub fn all_include_paths(&self) -> impl Iterator<Item = &PathBuf> {
+    pub fn all_include_paths(&self) -> Vec<&PathBuf> {
```

Comment:
If you don't want this API change, you could return `self.aggregate_path_buf(...).into_iter()` btw

Reply:
That would force to re-run a `collect` afterward, not sure if it's a good idea...

Reply:
Well like this it's technically an API change so it would have to become `system-deps` 4.0. It might be OK though because it only allows more things than before

Reply:
Keeping it as is then.